### DISCUSSION TITLE
Can a libtorrent programme know whether a torrent's webseed was used?

### DISCUSSION THREAD

**Discussion by cgdae** [arvidn/libtorrent#7803] [Q&A] [answered]
Apologies if this is a naive question but i'm just starting to use libtorrent and haven't been able to figure it out from the documentation.

I'd like to know whether my torrent's bep-0019 webseed was used when getting the data, or whether libtorrent has used bittorrent peers. I've enabled all alerts (except for the very verbose `dht_operation`), but they don't seem to show the information i'm looking for.

The torrent is for a file called `test.c`, and the alerts i get are:
```
Alert: torrent_added: test.c added
Alert: add_torrent: added torrent: test.c
Alert: state_changed: test.c: state changed to: checking
Alert: external_ip: external IP received: ##.##.##.##
Alert: torrent_resumed: test.c resumed
Alert: file_completed: test.c: file 0 finished downloading
Alert: state_changed: test.c: state changed to: finished
Alert: torrent_finished: test.c torrent finished downloading
Alert: state_changed: test.c: state changed to: seeding
```
Is there information available saying which pieces (if any) have been obtained from a torrent's webseed?

Thanks in advance for any help.

**Comment by arvidn** [comment, marked as answer]
you could ask for all the peers ([get_peer_info()](https://libtorrent.org/reference-Torrent_Handle.html#get-peer-info-post-peer-info)). In each of the peer's [peer_info](https://libtorrent.org/reference-Core.html#peer_info) object, you can look at the `connection_type` field to see if the `web_seed` or `http_seed` bits are set.

**Comment by cgdae** [comment]
Thanks for answering my question, i'll close this discussion now.
